roottest/root/io/perf/ReadBuffer.C

```diff
@@ -32,135 +32,52 @@ public:
 };
 
 
-void ReadBuffer(int siz=10) {
+void ReadBufferInt(int siz=10) {
 
-   TBuffer b(TBuffer::kWrite,32000);
+   TBufferFile b(TBuffer::kWrite,32000);
 
    allint obj;
    TClass *cl = gROOT->GetClass(typeid(obj));
    cl->Streamer(&obj,b);
 
    b.SetReadMode();
-   
+
    for(int i=0; i<siz; ++i) {
       b.Reset();
       cl->Streamer(&obj,b);
    }
 
 }
 
-void ReadBufferMix(int siz=10) {
+void ReadBufferFloat(int siz=10) {
 
-   TBuffer b(TBuffer::kWrite,32000);
+   TBufferFile b(TBuffer::kWrite,32000);
 
    fltint obj;
    TClass *cl = gROOT->GetClass(typeid(obj));
    cl->Streamer(&obj,b);
-   
-   b.SetReadMode();
-   
-   for(int i=0; i<siz; ++i) {
-      b.Reset();
-      cl->Streamer(&obj,b);
-   }
-
-}
-
-void InfoReadBuffer(int siz=10) {
-
-   TBuffer b(TBuffer::kWrite,32000);
 
-   allint obj;
-#if ROOT_VERSION_CODE<= 199169
-
-#else
-   char *pointer = (char*)&obj;
-#endif
-   TClass *cl = gROOT->GetClass(typeid(obj));
-   TStreamerInfo *info = cl->GetStreamerInfo();
-
-#if ROOT_VERSION_CODE<= 199169
-      info->WriteBuffer(b, (char*)(&obj),-1);
-#else
-      info->WriteBufferAux(b, &pointer,-1, 1, 0, 0);
-#endif
-   
    b.SetReadMode();
-   
-   for(int i=0; i<siz; ++i) {
-      b.Reset();
-      // cl->Streamer(&obj,b);
-#if ROOT_VERSION_CODE<= 199169
-      info->ReadBuffer(b, (char*)(&obj),-1);
-#else
-      info->ReadBuffer(b, (char*)(&obj),-1, 1, 0, 0);
-#endif
-   }
-
-}
-
-void InfoReadBufferMix(int siz=10) {
-
-   TBuffer b(TBuffer::kWrite,32000);
 
-   fltint obj;
-#if ROOT_VERSION_CODE<= 199169
-
-#else
-   char *pointer = (char*)&obj;
-#endif
-   TClass *cl = gROOT->GetClass(typeid(obj));
-   TStreamerInfo *info = cl->GetStreamerInfo();
-   
-#if ROOT_VERSION_CODE<= 199169
-      info->WriteBuffer(b, (char*)(&obj),-1);
-#else
-      info->WriteBufferAux(b, &pointer,-1, 1, 0, 0);
```

Comment:
This function is still public ...

Reply:
Yes, but one of the arguments `TCompInfo` is not accessible. 
And I am not sure if I can put nullptr instead.
So I did not find reasonable set of arguments to let it run.
Also there is no public `ReadBufferAux` to make sense test of writing.

Reply:
> Also there is no public ReadBufferAux to make sense test of writing.

I don't remember why but the converse of `WriteBufferAux` is simply `ReadBuffer`.

> Yes, but one of the arguments TCompInfo is not accessible.

Indeed :( ... It looks like 4e9c5a926b9067487f088b5cd80c4e2c3d2ba244 should have made `WriteBufferAux` private.  That commit is part of a large effort to migrate toward using StreamerInfoActions ...  Those performance test were part of making the decision to select the StreamerInfoAction solution ... so indeed they might be obsolete.